### PR TITLE
Make it possible for themes to blur backgrounds

### PR DESCRIPTION
**Pull request by Nezz**
## Proposed change

Currently themes have little control over the `backdrop-filter` properties. The only one they can control is the whole screen when a dialog is open:
https://github.com/home-assistant/frontend/blob/dd082c204b45df55663dc257ee83f5efe1a9ab88/src/components/ha-dialog.ts#L78-L79

This PR makes it possible for themes to control the `backdrop-filter` for cards and dialogs. This allows the use of blurred background among others. For example without and with a blur filter:

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```css
--ha-card-background: rgba(0, 0, 0, 0.1);
--ha-card-backdrop-filter: blur(20px);
--dialog-surface-backdrop-filter: blur(20px);
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
https://github.com/basnijholt/lovelace-ios-themes/issues/61
https://github.com/basnijholt/lovelace-ios-themes/issues/49
https://github.com/basnijholt/lovelace-ios-themes/issues/22
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by basnijholt** [comment]
This would be a great addition and definitely worth it regarding the small change set!

File: src/components/ha-dialog.ts [hunk 1]
```diff
@@ -119,6 +119,8 @@ export class HaDialog extends DialogBase {
         margin-top: var(--dialog-surface-margin-top);
         min-height: var(--mdc-dialog-min-height, auto);
         border-radius: var(--ha-dialog-border-radius, 28px);
+        -webkit-backdrop-filter: var(--dialog-surface-backdrop-filter, none);
+        backdrop-filter: var(--dialog-surface-backdrop-filter, none);
```
Comment (bramkragten) [inline src/components/ha-dialog.ts:123]:
```suggestion
        -webkit-backdrop-filter: var(--ha-dialog-surface-backdrop-filter, none);
        backdrop-filter: var(--ha-dialog-surface-backdrop-filter, none);
```

**Comment by Nezz** [comment]
Good point. Should it actually be this?
```
var(--ha-dialog-scrim-backdrop-filter, var(--dialog-backdrop-filter, none));
```

**Comment by bramkragten** [comment]
> Good point. Should it actually be this?
> 
> ```
> var(--ha-dialog-scrim-backdrop-filter, var(--dialog-backdrop-filter, none));
> ```

The default is none anyway, so not really needed but sure 👍

**Review by bramkragten — APPROVED** [review]

File: src/components/ha-dialog.ts [hunk 2]
```diff
@@ -119,6 +125,8 @@ export class HaDialog extends DialogBase {
         margin-top: var(--dialog-surface-margin-top);
         min-height: var(--mdc-dialog-min-height, auto);
         border-radius: var(--ha-dialog-border-radius, 28px);
+        -webkit--filter: var(--ha-dialog-surface--filter, none);
+        -filter: var(--ha-dialog-surface--filter, none);
```
Comment (RhavoX) [inline src/components/ha-dialog.ts:129]: Shouldn't this be a single dash here: --ha-dialog-surface--filter?

Comment (Nezz) [reply src/components/ha-dialog.ts:129]: Good point. I fixed this in my fork but for some reason not on this branch.

Comment (ildar170975) [reply src/components/ha-dialog.ts:129]: So, it left with a double dash?

**Pull request merged**